channeldb/payment_control.go

```diff
@@ -223,6 +223,17 @@ func (p *PaymentControl) InitPayment(paymentHash lntypes.Hash,
 	return updateErr
 }
 
+// DeleteFailedAttempts deletes all failed htlcs for a payment.
+func (p *PaymentControl) DeleteFailedAttempts(hash lntypes.Hash) error {
+	if !p.db.keepFailedPaymentAttempts {
+		err := p.db.DeletePayment(hash, true)
```

Comment:
nit: maybe worth adding a const here to name what true means in this context (for readability).

Reply:
Done - at least I think. Did I do what you were asking me to do?